suuchi-core/src/main/scala/in/ashwanthkumar/suuchi/store/InMemoryStore.scala

```diff
@@ -26,4 +30,19 @@ class InMemoryStore extends Store {
     store.remove(ByteBuffer.wrap(key))
     true
   }
+
+  override def scan(): Iterator[KV] = {
+    store.entrySet().map(kv => KV(kv.getKey.array(), kv.getValue)).iterator
+  }
+
+  private def hasPrefix(key: Array[Byte], prefix: Array[Byte]) = {
```

Comment:
@gsriram7 This looks identical to one on [RocksDBStore](https://github.com/ashwanthkumar/suuchi/pull/61/files#diff-1fe1ed88e9d16ea01283742388c4c6e1R51). Can you see if you can extract this out as a reusable method inside an `Object` ?